servers/physics_2d/shape_2d_sw.cpp

```diff
@@ -637,7 +634,7 @@ real_t ConvexPolygonShape2DSW::get_moment_of_inertia(real_t p_mass, const Size2
 		aabb.expand_to(points[i].pos * p_scale);
 	}
 
-	return p_mass * aabb.size.dot(aabb.size) / 12.0 + p_mass * (aabb.position + aabb.size * 0.5).length_squared();
```

Comment:
Are you sure about removing the offset for the center of gravity in this and SegmentShape2D? The formulas on Wikipedia are given for an axis passing through the center of gravity, but maybe that's not what we take as referential in Godot?

Reply:
I'm not quite familiar with it. But I think if we need a moment of inertia with an axis not passing center of gravity, we should recalculate it with origin result, there is already it: 				https://github.com/godotengine/godot/blob/252c841d7ff5d770e8d2819e4f7955363410744b/servers/physics_2d/body_2d_sw.cpp#L76
It seems unnecessary to add offset here.

Reply:
Looks like it indeed.